app/js/view/details_view.js

```diff
@@ -35,14 +39,75 @@ export default class DetailsView extends View {
     });
   }
 
+  filterNonAffectedApps(apps, filters) {
+    return apps.filter(app => {
+      return filters.reduce((prev, cur) => {
+        return prev || cur.test(app.manifestURL);
+      }, false);
+    });
+  }
+
+  populateAffectedApps(details) {
+    Promise.all([
+      AppsHelper.getAllApps(),
+      ManifestHelper.getManifest(details.manifestURL)
+    ])
+
+    .then(results => {
```

Comment:
Ok, I feel like it's more readable, but I guess it's against convention? I'm still figuring promises out.